v2/pkg/engine/datasource/pubsub_datasource/pubsub_datasource_test.go

```diff
@@ -53,32 +49,48 @@ func TestPubSub(t *testing.T) {
 	dataSourceCustomConfig := Configuration{
 		Events: []EventConfiguration{
 			{
-				FieldName:  "helloQuery",
-				SourceName: "default",
-				Subjects:   []string{"helloQuery.{{ args.id }}"},
-				Type:       EventTypeRequest,
-				TypeName:   "Query",
+				Metadata: &EventMetadata{
+					ProviderID: "default",
+					FieldName:  "helloQuery",
+					Type:       EventTypeRequest,
+					TypeName:   "Query",
+				},
+				Configuration: &NatsEventConfiguration{
+					Subjects: []string{"helloQuery.{{ args.id }}"},
```

Comment:
The EDFS spec allows patterns like `helloQuery.{{ args.input.id }}` where the field argument has a root arg (input) with an input type that contains a field "id". Is this also supported by your implementation?

Reply:
Everything works as before in terms of NATS. I didn't touch this. I focused on the abstraction. Kafka won't support pattern in topics due to its different model.